### PR TITLE
feat: add clone environment api

### PR DESCRIPTION
**Pull request by shanewxy**
**Problem:**

The original API used for cloning environment is the same as creating environment, which cannot handle the case of cloning sensitive variables.

**Solution:**

Add a specific clone environment API to decouple with the creation API, which queries and fills in the value of sensitive variables.

**Related Issue:**
#985

### CODE REVIEW

File: pkg/apis/environment/extension.go [hunk 1]
```diff
@@ -150,3 +153,25 @@ func getCaps(entities model.Services) (int, int) {
...
+	entity := req.Model()
+
+	for i := range req.Variables {
+		v := entity.Edges.Variables[i]
+		if v == nil {
+			return nil, errorx.New("invalid input: nil variable")
+		}
+
+		if v.Sensitive && v.Value == "" {
+			v.Value = h.modelClient.Variables().Query().
```
Comment (gitlawr) [inline pkg/apis/environment/extension.go:167]: Can we query vars once instead of querying one for each var?

Comment (shanewxy) [reply pkg/apis/environment/extension.go]: updated

**Review by gitlawr — APPROVED** [review]
LGTM

File: pkg/apis/environment/extension.go [hunk 2]
```diff
@@ -150,3 +154,44 @@ func getCaps(entities model.Services) (int, int) {
...
+	}
+
+	// Fetch and fill the value of sensitive variables from the cloned environment.
+	if len(variableNames) > 0 {
+		vs := h.modelClient.Variables().Query().
+			Where(
+				variable.EnvironmentID(req.CloneEnvironmentId),
+				variable.NameIn(variableNames...),
+			).
+			AllX(req.Context)
```
Comment (alexcodelf) [inline pkg/apis/environment/extension.go:182]: using All(req.Context) instead of AllX

File: pkg/apis/environment/extension_view.go [hunk 3]
```diff
@@ -27,3 +33,91 @@ type (
...
+
+		CloneEnvironmentId object.ID `path:"environment"`
+
+		model.EnvironmentCreateInput `path:",inline" json:",inline"`
+	}
+
+	RouteCloneEnvironmentResponse = model.EnvironmentOutput
+)
+
+func (r *RouteCloneEnvironmentRequest) Validate() error {
```
Comment (alexcodelf) [inline pkg/apis/environment/extension_view.go:49]: It seems that the clone request validate is similar to the create environment validate, can you using a common validate func for two request validation?

File: pkg/apis/environment/extension.go (hunk 2 above)
Comment (shanewxy) [reply pkg/apis/environment/extension.go]: updated

File: pkg/apis/environment/extension_view.go (hunk 3 above)
Comment (shanewxy) [reply pkg/apis/environment/extension_view.go:44]: updated

**Review by alexcodelf — APPROVED** [review]
LGTM

**Pull request merged**